qemu/tests/uefi_check_debugcon.py

```diff
@@ -0,0 +1,118 @@
+        """
+        process.system(trace_record_cmd)
+
+    error_context.context("Check whether trace-cmd has been "
+                          "installed on host.", logging.info)
+    if process.system("which trace-cmd", ignore_status=True):
+        test.error("Please install trace-cmd on host.")
+    if params.get("ovmf_log"):
+        error_context.context("Append debugcon parameter to "
+                              "qemu command lines.", logging.info)
+        ovmf_log = utils_misc.get_path(test.debugdir, params["ovmf_log"])
+        params["extra_params"] %= ovmf_log
+        params["start_vm"] = "yes"
+        env_process.process(test, params, env,
+                            env_process.preprocess_image,
+                            env_process.preprocess_vm)
+    trace_output_file = utils_misc.get_path(test.debugdir,
+                                            params["trace_output"])
+    trace_record_cmd = params["trace_record_cmd"] % trace_output_file
+    check_pio_read = params["check_pio_read"] % trace_output_file
+    check_pio_write = params["check_pio_write"] % trace_output_file
+    stop_trace_record = params["stop_trace_record"]
+    timeout = int(params.get("timeout", 120))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    error_context.context("Remove the existing isa-log device.", logging.info)
+    remove_isa_debugcon(vm)
+    vm.destroy()
+    error_context.context("Run trace record command on host.",
+                          logging.info)
+    bg = utils_test.BackgroundTest(trace_kvm_pio, ())
+    bg.start()
+    if not utils_misc.wait_for(lambda: bg.is_alive, timeout):
+        test.fail("Failed to start command: '%s'" % trace_record_cmd)
+    try:
+        vm.create()
+        vm.verify_alive()
+        vm.destroy()
+        process.system(stop_trace_record, ignore_status=True, shell=True)
+        if not utils_misc.wait_for(
+                lambda: not check_trace_process(), timeout, 30, 3):
+            test.fail("Failed to stop command: '%s' after %s seconds."
+                      % (stop_trace_record, timeout))
+        pio_read_counts = int(process.run(
+            check_pio_read, shell=True).stdout.decode().strip())
+        err_str = "pio_read counts should be greater than 0. "
+        err_str += "But the actual counts are %s." % pio_read_counts
+        test.assertGreater(pio_read_counts, 0, err_str)
+        pio_write_counts = int(process.run(
+            check_pio_write, shell=True).stdout.decode().strip())
+        if params.get("ovmf_log"):
+            err_str = "pio_write counts should be greater than 0. "
+            err_str += "But the actual counts are %s." % pio_write_counts
+            test.assertGreater(pio_write_counts, 0, err_str)
+        else:
+            err_str = "pio_write counts should be equal to 0. "
+            err_str += "But the actual counts are %s." % pio_write_counts
+            test.assertEqual(pio_write_counts, 0, err_str)
+    finally:
```

Comment:
remove ovmf_log and trace_output_file if they exist

Reply:
Don't remove them, put them into test.debugdir. It's a part of logs.  thanks